src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -432,6 +433,36 @@ void WiFiManager::ConnectHandler(Platform::UniquePtr<uint8_t> data)
             {
                 Instance().mHandling.mOnConnectionDone(connStatus);
             }
+
+            WiFiDiagnosticsDelegate * delegate = GetDiagnosticDataProvider().GetWiFiDiagnosticsDelegate();
+            if (delegate)
+            {
+                uint16_t reason = Instance().GetLastDisconnectReason();
+                uint8_t associationFailureCause;
+
+                switch (connStatus)
+                {
+                case WIFI_STATUS_CONN_WRONG_PASSWORD:
+                    associationFailureCause = chip::to_underlying(
+                        chip::app::Clusters::WiFiNetworkDiagnostics::AssociationFailureCauseEnum::kAuthenticationFailed);
```

Comment:
Might be worth it to have a `using app::Clusters::WiFiNetworkDiagnostics::AssociationFailureCauseEnum` at the top of this function so these bits can be shorter.